funnel/forms/space.py

```diff
@@ -92,6 +93,30 @@ class ParticipantForm(forms.Form):
         validators=[forms.validators.DataRequired(u"Select at least one event")])
 
 
+# TODO: Move to Baseframe
+class ValidFile(object):
+    def __init__(self, allowed_exts=[], message=None):
+        if not message:
+            message = __(u"Please upload a valid file.")
+        self.message = message
+        self.allowed_exts = allowed_exts
+
+    def __call__(self, form, field):
+        ext = os.path.splitext(field.data.filename)[1].strip(".")
+        if ext and ext.lower() not in self.allowed_exts:
+            raise wtforms.validators.StopValidation(self.message)
```

Comment:
Any file passed through MS Excel on Mac will have CR for linefeeds, not LF or CRLF, and will be in Latin1 charset instead of UTF-8. You need to detect that and massage the data.